### PR TITLE
Batch trivialize unnecessary cloudfront.net rewrites

### PR DESCRIPTION
**Pull request by cschanaj**
Attached the source code used to perform the batch rewrites

- https://gist.github.com/cschanaj/31abc5c1f08416683aeb54e74d3846ff

### CODE REVIEW

File: src/chrome/content/rules/CKSource.com.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
 
 
 	<rule from="^http://c\.cksource\.com/"
-		to="https://d2xc7e2akbvihw.cloudfront.net/" />
+		to="https://c.cksource.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/CKSource.com.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/ComiXology.com.xml [hunk 2]
```diff
@@ -87,7 +87,7 @@
 		to="https://www.comixology.com/" />
 
 	<rule from="^http://support\.comixology\.com/"
-		to="https://d3jyn100am7dxp.cloudfront.net/" />
+		to="https://support.comixology.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/ComiXology.com.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/DealerRater.com.xml [hunk 3]
```diff
@@ -28,9 +28,9 @@
 		to="https://$1dealerrater.com/" />
 
 	<rule from="^http://cdn-static\.dealerrater\.com/"
```
Comment (Bisaloo) [inline src/chrome/content/rules/DealerRater.com.xml]: This subdomain is not listed in the targets

File: src/chrome/content/rules/DealerRater.com.xml [hunk 4]
```diff
@@ -28,9 +28,9 @@
...
-		to="https://d3go5n7g2hd1g0.cloudfront.net/" />
+		to="https://cdn-static.dealerrater.com/" />
 
 	<rule from="^http://cdn-user\.dealerrater\.com/"
```
Comment (Bisaloo) [inline src/chrome/content/rules/DealerRater.com.xml]: This subdomain is not listed in the targets

File: src/chrome/content/rules/Filemobile.xml [hunk 5]
```diff
@@ -44,7 +44,7 @@
 		to="https://s3.amazonaws.com/$1.filemobile.com/" />
 
 	<rule from="^http://cimg\.filemobile\.com/"
-		to="https://d2ksawk1637r8x.cloudfront.net/" />
+		to="https://cimg.filemobile.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/Filemobile.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/GCaptain.com.xml [hunk 6]
```diff
@@ -70,7 +70,7 @@ Non-2xx HTTP code: http://cfs1.gcaptain.com/ (200) => https://d2m8pnbf2v4ae2.clo
 		to="https://d38ecmhxsvwui3.cloudfront.net/" />
 
 	<rule from="^http://cfs1\.gcaptain\.com/"
-		to="https://d2m8pnbf2v4ae2.cloudfront.net/" />
+		to="https://cfs1.gcaptain.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/GCaptain.com.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/MESO-Rx.xml [hunk 7]
```diff
@@ -47,7 +47,7 @@ Non-2xx HTTP code: http://www.thinksteroids.com/ (200) => https://www.thinkstero
 
 
 	<rule from="^http://cdn\.thinksteroids\.com/"
-		to="https://d1uxx6tvzy89fj.cloudfront.net/" />
+		to="https://cdn.thinksteroids.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/MESO-Rx.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/Rovio.com.xml [hunk 8]
```diff
@@ -65,7 +65,7 @@
 		<test url="http://assets.rovio.com/rovio/favicon.ico" />
 
 	<rule from="^http://assets-production\.rovio\.com/"
-		to="https://ductl1gjw76cl.cloudfront.net/" />
+		to="https://assets-production.rovio.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/Rovio.com.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/Sailthru.xml [hunk 9]
```diff
@@ -112,7 +112,7 @@ Fetch error: http://me.sailthru.com/ => https://www.sailthru.com/: Too many redi
 		to="https://na-sj08.marketo.com/" />
 
 	<rule from="^http://media\.sailthru\.com/"
-		to="https://d1t6td7a2qcurl.cloudfront.net/" />
+		to="https://media.sailthru.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/Sailthru.xml]: This rule can be removed (covered by the trivial rewrite)

File: src/chrome/content/rules/scmagazine.com.xml [hunk 10]
```diff
@@ -60,7 +60,7 @@
 		to="https://www.scmagazine.com/" />
 
 	<rule from="^http://media\.scmagazine\.com/"
-		to="https://dw5yb2c18zulu.cloudfront.net/" />
+		to="https://media.scmagazine.com/" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/scmagazine.com.xml]: This rule can be removed (covered by the trivial rewrite)

**Comment by cschanaj** [comment]
@Bisaloo I have removed the redundant rules and addressed the missing targets as well. It seems that `DealerRater.com.xml` require a bit more modifications. Travis is passing now. Thank you very much!!

**Comment by J0WI** [comment]
I'm not so familiar with nodejs, but I think this is a valuable enhancement.
How did you find the hosts in `list.js`?

**Comment by Hainish** [comment]
In general ruleset parsing scripts should be checked in, along with the methodology to generate any data sets that the parsing script relies on (such as the hosts in `list.js`).

Some of the host clean-up scripts that @RReverser has run haven't been checked in, but I think this is a good practice in general, and we may want to start a `util/` just for this.

**Comment by RReverser** [comment]
> Some of the host clean-up scripts that @RReverser has run haven't been checked in

I think I have checked in all of them, which one you have in mind?

**Comment by RReverser** [comment]
Looking at the diff in this PR, many of these should be possible to actually trivialise further by running `node utils/trivialize-rules` on top of these changes. @cschanaj would you want to try that?

**Comment by Bisaloo** [comment]
@RReverser, it looks like this was done already in https://github.com/EFForg/https-everywhere/pull/15998/commits/61cec8e297448fdc5c838ca20c88acff5316cd7f. Do you think the subsequent commits allow for further simplification by the script?

**Comment by RReverser** [comment]
@Bisaloo Ah, I've just seen Adk2.xml which can be trivialised further, but it can't by automatic script as there is `securecookie` tag (which might be used for cookie on other subdomains).

**Comment by cschanaj** [comment]
@J0WI @Hainish the script I used to find the host is rather dirty and include a bit of manual steps.

1. `./make.sh`
2. `node index.js > input.txt`
3. `cat input.txt | while read host; do curl -IsS https://$host/ -o /dev/null && echo $host; done > success.txt`
4. manual changes to `success.txt` to `list.json`

https://gist.github.com/cschanaj/2ee89cb791eeda0e764b167c5c3f4ede

@RReverser I agree that some rulesets can be futher trivialized but the `securecookie`  tags forbidden that. I tried to make some changes (still work in progress) to the script to identify static cookies and it seem to work. I can submit a PR later if I can verify the modified script is really working. https://github.com/cschanaj/https-everywhere/blob/trivialize-rules.js/utils/trivialize-rules/trivialize-rules.js

**Comment by Hainish** [comment]
@cschanaj it's okay if the scripts are sloppy, I think the important part is that the process is documented and can be reproduced in the future.  If someone comes along and cleans it up (or you have time to down the line), all the better.

**Comment by Bisaloo** [comment]
@cschanaj, it would be great if you could add your script to the repo as suggested by @Hainish.

Otherwise, this is ready to merge.

**Comment by cschanaj** [comment]
@Bisaloo @Hainish I have spent some time rewriting the script in node. While I did not check if it will reproduce the same rewrites as in this PR, it should be generic enough for future runs. Thanks!!

**Comment by cschanaj** [comment]
@Bisaloo @Hainish I have pushed a final version of the script in 87a9087. Sorry for force pushing a few time. This PR is ready to merge if you agree.

**Comment by Bisaloo** [comment]
Can I edit the commit message to remove intermediate commits and just keep `Batch trivialize unnecessary cloudfront.net rewrites` when I squash and merge or do you wish to keep all intermediate steps?

**Comment by cschanaj** [comment]
@Bisaloo please feel free to edit the commit message. thanks!!

**Comment by Hainish** [comment]
Thanks for this PR @cschanaj and for the review @Bisaloo!

**Pull request merged**